src/scenes/levelend-scene/levelend-scene.ts

```diff
@@ -142,6 +118,70 @@ export class LevelEndScene {
     }
   }
 
+  private createButton(
+    ButtonClass: typeof MapButton | typeof RetryButtonHtml | typeof NextButtonHtml,
+    id: string,
+    onClickCallback: () => void
+  ) {
+    const buttonsContainerId = 'buttons-container';
+
+    const button = new ButtonClass({ targetId: buttonsContainerId, id });
+    const gameControl = document.getElementById("game-control") as HTMLCanvasElement;
+    button.onClick(() => {
+      this.levelEndElement.style.zIndex = '7';
+      gameControl.style.zIndex = "-1";
+      onClickCallback();
+    });
+  };
+
+  renderButtonsHTML() {
+    // Define configurations for each button
+    const buttonConfigs = [
+      {
+        ButtonClass: MapButton,
+        id: 'levelend-map-btn',
+        onClick: () => {
+          this.handlePublishEvent(true);
+          this.switchToLevelSelectionCB();
```

Comment:
Same comment with the reusable method for publishing, I think `this.handlePublishEvent(true)` on callback can be simplified within the three callback config you created here.

Reply:
@bernhardccodev thank you for your suggestion. i applied an update here. thanks